v2/audiences.go

```diff
@@ -0,0 +1,193 @@
+package courier
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"net/http"
+)
+
+type SingleFilter struct {
+	Operator string `json:"operator"`
+	Path     string `json:"path"`
+	Value    string `json:"value"`
+}
+
+type NestedFilter struct {
+	Operator string         `json:"operator"`
+	Rules    []SingleFilter `json:"rules"`
+}
+
+type Audience struct {
+	Description string      `json:"description,omitempty"`
+	Filter      interface{} `json:"filter"`
+	Name        string      `json:"name,omitempty"`
+}
+
+type AudienceResponseBody struct {
+	Id          string      `json:"id"`
+	Description string      `json:"description"`
+	CreatedAt   string      `json:"created_at"`
+	Filter      interface{} `json:"filter"`
+	Name        string      `json:"name"`
+	UpdatedAt   string      `json:"updated_at"`
+}
+
+type AudienceResponse struct {
+	Audience AudienceResponseBody `json:"audience"`
+}
+
+type AudienceMember struct {
+	AudienceId      string `json:"audience_id"`
+	AddedAt         string `json:"added_at"`
+	AudienceVersion int    `json:"audience_version"`
+	MemberId        string `json:"member_id"`
+	Reason          string `json:"reason"`
+}
+
+type GetAudienceMembersResponse struct {
+	Items  []*AudienceMember `json:"items"`
+	Paging *PagingResponse   `json:"paging"`
+}
+
+type GetAudiencesResponse struct {
+	Items  []*AudienceResponseBody `json:"items"`
+	Paging *PagingResponse
+}
+
+func (c *Client) PutAudience(ctx context.Context, audienceId string, audience Audience) (*AudienceResponse, error) {
+	if audienceId == "" {
+		return nil, errors.New("Audience ID is required")
+	}
+
+	_, singleFilterOk := audience.Filter.(SingleFilter)
+	_, nestedFilterOk := audience.Filter.(NestedFilter)
+
+	if !singleFilterOk && !nestedFilterOk {
+		return nil, errors.New("Audience filter must be a SingleFilter or NestedFilter")
```

Comment:
reword error message to the following perhaps`Audience filter must be of type SingleFIlter or NestFilter`

Reply:
```suggestion
		return nil, errors.New("Audience filter must be of type SingleFilter or NestFilter")
```